amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -207,6 +209,10 @@ public void run() {
 
         finalShutdown();
         log.info("Worker loop is complete. Exiting from worker.");
+
+        if (exceptionToThrow != null) {
```

Comment:
I don't feel this is the correct location to report the exception.  We should report as near as possible to where the exception originally occurred.  This would be important if the user was also attempting to create a stack trace for the original call site.